src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -1057,6 +1059,16 @@ QString QgsPostgresConn::quotedValue( const QVariant &value )
   }
 }
 
+QString QgsPostgresConn::quotedJsonValue( const QVariant &value )
+{
+  if ( value.isNull() || !value.isValid() )
```

Comment:
```suggestion
  if ( value.isNull() || !value.isValid() || value.type() == QVariant::Bool )
     return QgsPostgresConn::quotedValue( value );
```

(for consistency)?

Reply:
I've added a few more tests, null, true and false should be handled correctly now.